src/lib/controller/incoming_message.ts

```diff
@@ -243,6 +243,7 @@ export interface IncomingCommandControllerGetAvailableFirmwareUpdates
   command: ControllerCommand.getAvailableFirmwareUpdates;
   nodeId: number;
   apiKey?: string;
+  additionalUserAgentComponents?: Record<string, string>;
```

Comment:
Wouldn't it make more sense to add this to the initial handshake? That way it is set once and can be used for other methods too.